File: smartexpense-server/src/main/java/com/smartexpense/server/dto/AuthRequest.java
package com.smartexpense.server.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

/**
 * Data Transfer Object (DTO) chứa thông tin đăng nhập hoặc đăng ký tài khoản từ Client gửi lên.
 */
@Data
public class AuthRequest {
    
    @Email(message = "Email không đúng định dạng")
    @NotBlank(message = "Email không được để trống")
    private String email; // Địa chỉ email tài khoản

    @NotBlank(message = "Mật khẩu không được để trống")
    private String password; // Mật khẩu người dùng dạng thô

    private String name; // Họ tên đầy đủ người dùng (chỉ gửi lên khi đăng ký)
}

File: app/src/main/java/com/arijit/budgettracker/api/AuthRequest.kt
package com.arijit.budgettracker.api

/**
 * Lớp dữ liệu (Data Class) chứa thông tin yêu cầu xác thực gửi từ Client lên Server.
 * Dùng chung cho cả hai nghiệp vụ Đăng ký tài khoản mới và Đăng nhập.
 *
 * @property email Địa chỉ email của người dùng (dùng làm tên đăng nhập chính).
 * @property password Mật khẩu đăng nhập dạng thô.
 * @property name Họ và tên đầy đủ của người dùng (chỉ bắt buộc khi Đăng ký).
 * @property phone Số điện thoại của người dùng (tùy chọn).
 * @property locked Trạng thái khóa tài khoản (mặc định là 0 - không khóa).
 * @property role Vai trò phân quyền trong hệ thống (mặc định gửi "USER").
 */
data class AuthRequest(
    val email: String,
    val password: String,
    val name: String? = null,
    val phone: String? = null,
    val locked: Int = 0,
    val role: String = "USER"
)